multimap/src/main/java/org/infinispan/multimap/impl/SortedSetBucket.java

```diff
@@ -0,0 +1,208 @@
+package org.infinispan.multimap.impl;
+
+import org.infinispan.commons.marshall.ProtoStreamTypeIds;
+import org.infinispan.multimap.impl.internal.MultimapObjectWrapper;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.protostream.annotations.ProtoTypeId;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+/**
+ * Bucket used to store Sorted Set data type.
+ *
+ * @author Katia Aresti
+ * @since 15.0
+ */
+@ProtoTypeId(ProtoStreamTypeIds.MULTIMAP_SORTED_SET_BUCKET)
+public class SortedSetBucket<V> {
+   private final TreeSet<ScoredValue<V>> scoredEntries;
+   private final TreeSet<ScoredValue<V>> lexicalOrderEntries;
```

Comment:
the way it works, we can't find in a tree set something by score or try to check if a value exists. it will navigate through the tree
moreover, when looking to other commands in sorted set, you can ask for retrieving in lexical or score ordering, which means we need to hold the values in different structures.
Redis implementation is a double structure as well, for these reasons